Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/MarkerDetection/QRCode/QRCodeMarkerDetector.cs

```diff
@@ -0,0 +1,257 @@
+// to the general public until said distribution story is determiend. However, this file has been added to enable
+// public facing development.
+// #define QRCODESTRACKER_BINARY_AVAILABLE
+
+#if QRCODESTRACKER_BINARY_AVAILABLE
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.QRCodesTracker;
+#endif
+
+using UnityEngine;
+
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.MarkerDetection;
+using System.Collections.Generic;
+using System;
+
+#if UNITY_WSA && WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+using Windows.Perception.Spatial;
+using Windows.Perception.Spatial.Preview;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView.MarkerDetection
+{
+    /// <summary>
+    /// QR code detector that implements <see cref="Microsoft.MixedReality.Toolkit.Extensions.MarkerDetection.IVariableSizeMarkerDetector"/>
+    /// </summary>
+    public class QRCodeMarkerDetector : MonoBehaviour,
+        IVariableSizeMarkerDetector
+    {
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+        private QRCodesManager _qrCodesManager;
+        private Dictionary<Guid, SpatialCoordinateSystem> _markerCoordinateSystems = new Dictionary<Guid, SpatialCoordinateSystem>();
+#endif
+        private class QRCodeInfo
+        {
+            public Guid QRCodeId;
+            public int MarkerId;
+            public float MarkerSize;
+            public QRCodeInfo(Guid qrCodeId, int markerId, float markerSize)
+            {
+                this.QRCodeId = qrCodeId;
+                this.MarkerId = markerId;
+                this.MarkerSize = markerSize;
+            }
+        }
+
+        private bool _processMarkers = false;
+        private Dictionary<Guid, QRCodeInfo> _cachedMarkers = new Dictionary<Guid, QRCodeInfo>();
+        private Dictionary<int, List<Marker>> _markerObservations = new Dictionary<int, List<Marker>>();
+        private readonly string _qrCodeNamePrefix = "sv";
+
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+        private bool _tracking = false;
+#endif
+
+#pragma warning disable 67
+        /// <inheritdoc />
+        public event MarkersUpdatedHandler MarkersUpdated;
+#pragma warning restore 67
+
+        /// <inheritdoc />
+        public void SetMarkerSize(float markerSize){}
+
+        /// <inheritdoc />
+        public void StartDetecting()
+        {
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+            _tracking = true;
+#else
+            Debug.LogError("Current platform does not support qr code marker detector");
+#endif
+        }
+
+        /// <inheritdoc />
+        public void StopDetecting()
+        {
+#if WINDOWS_UWP && QRCODESTRACKER_BINARY_AVAILABLE
+            _tracking = false;
+            _processMarkers = false;
+#else
+            Debug.LogError("Current platform does not support qr code marker detector");
+#endif
+        }
+
+        /// <inheritdoc />
+        public bool TryGetMarkerSize(int markerId, out float size)
```

Comment:
Is this a texture representation of a Guid? Can we parse it and do and do a dictionary lookup?

Reply:
I'm not sure if this comment is at the correct location. Currently this just enables a marker detector consumer to provide a marker id and get back a size. The marker id is currently an integer. It doesn't have a determinable mapping to a guid.